x-pack/legacy/plugins/lens/public/app_plugin/plugin.tsx

```diff
@@ -93,9 +104,54 @@ export class AppPlugin {
             http: core.http,
           })
         );
+        const updateUrlTime = (urlVars: Record<string, string>): void => {
+          const decoded: RisonObject = rison.decode(urlVars._g) as RisonObject;
+          if (!decoded) {
+            return;
+          }
+          // @ts-ignore
+          decoded.time = data.query.timefilter.timefilter.getTime();
+          urlVars._g = rison.encode((decoded as unknown) as RisonObject);
+        };
+        const redirectTo = (
+          routeProps: RouteComponentProps<{ id?: string }>,
+          addToDashboardMode: boolean,
+          id?: string
+        ) => {
+          if (!id) {
+            routeProps.history.push('/lens');
+          } else if (!addToDashboardMode) {
+            routeProps.history.push(`/lens/edit/${id}`);
+          } else if (addToDashboardMode && id) {
+            routeProps.history.push(`/lens/edit/${id}`);
+            const url = context.core.chrome.navLinks.get('kibana:dashboard');
+            if (!url) {
+              return;
+            }
+            const lastDashboardAbsoluteUrl = url.url;
+            const lensUrl = `${getKibanaBasePathFromDashboardUrl(
+              lastDashboardAbsoluteUrl
+            )}/lens/edit/${id}`;
+            if (lastDashboardAbsoluteUrl && lensUrl) {
+              const urlVars = getUrlVars(lastDashboardAbsoluteUrl);
+              updateUrlTime(urlVars);
+              window.history.pushState({}, '', lensUrl);
+              const dashboardUrl = getDashboardUrlWithQueryParams(
+                lastDashboardAbsoluteUrl,
+                urlVars
+              );
+              const dashboardParsedUrl = addEmbeddableToDashboardUrl(dashboardUrl, id, 'lens');
+              if (dashboardParsedUrl) {
```

Comment:
If this returns null, we silently fail. Is that the right thing? It seems like a null here is a legit error that should be surfaced in some way.